src/gpu_sysman.c

```diff
@@ -0,0 +1,1939 @@
+      ok = false;
+      break;
+    }
+
+    bool correctable;
+    uint64_t value, total = 0;
+    const char *catname, *help;
+    for (int cat_idx = 0; cat_idx < ZES_MAX_RAS_ERROR_CATEGORY_COUNT;
+         cat_idx++) {
+      value = values.category[cat_idx];
+      total += value;
+      if (gpu->disabled.ras_separate) {
+        continue;
+      }
+      correctable = true;
+      switch (cat_idx) {
+        // categories which are not correctable, see:
+        // https://spec.oneapi.io/level-zero/latest/sysman/PROG.html#querying-ras-errors
+      case ZES_RAS_ERROR_CAT_RESET:
+        help = "Total number of GPU reset attempts by the driver";
+        catname = METRIC_PREFIX "resets_total";
+        correctable = false;
+        break;
+      case ZES_RAS_ERROR_CAT_PROGRAMMING_ERRORS:
+        help = "Total number of non-correctable HW exceptions generated by the "
+               "way workloads have programmed the HW";
+        catname = METRIC_PREFIX "programming_errors_total";
+        correctable = false;
+        break;
+      case ZES_RAS_ERROR_CAT_DRIVER_ERRORS:
+        help = "total number of non-correctable low level driver communication "
+               "errors";
+        catname = METRIC_PREFIX "driver_errors_total";
+        correctable = false;
+        break;
+        // categories which can have both correctable and uncorrectable errors
+      case ZES_RAS_ERROR_CAT_COMPUTE_ERRORS:
+        help = "Total number of errors occurrend in the accelerator HW";
+        catname = METRIC_PREFIX "compute_errors_total";
+        break;
+      case ZES_RAS_ERROR_CAT_NON_COMPUTE_ERRORS:
+        help = "Total number of errors occurred in the fixed-function "
+               "accelerator HW";
+        catname = METRIC_PREFIX "fixed_function_errors_total";
+        break;
+      case ZES_RAS_ERROR_CAT_CACHE_ERRORS:
+        help = "Total number of ECC errors that have occurred in the on-chip "
+               "caches";
+        catname = METRIC_PREFIX "cache_errors_total";
+        break;
+      case ZES_RAS_ERROR_CAT_DISPLAY_ERRORS:
+        help = "Total number of ECC errors that have occurred in the display";
+        catname = METRIC_PREFIX "display_errors_total";
+        break;
+      default:
+        help = "Total number of errors in unsupported categories";
+        catname = METRIC_PREFIX "unknown_errors_total";
+      }
+      if (correctable) {
+        ras_submit(gpu, catname, help, type, subdev, value);
+      } else if (props.type == ZES_RAS_ERROR_TYPE_UNCORRECTABLE) {
+        ras_submit(gpu, catname, help, NULL, subdev, value);
+      }
+    }
+    catname = METRIC_PREFIX "all_errors_total";
+    help = "Total number of errors in all categories";
+    ras_submit(gpu, catname, help, type, subdev, total);
+    ok = true;
+  }
+  free(ras);
+  return ok;
+}
+
+static void metric_set_subdev(metric_t *m, bool onsub, uint32_t subid) {
+  if (onsub) {
+    char buf[8];
+    snprintf(buf, sizeof(buf), "%d", subid);
```

Comment:
https://github.com/collectd/collectd/blob/main/src/utils/common/common.c#L93

Reply:
Nope, as snprintf -> ssnprintf is pointless change, and because all used collectd functions need to be copied gpu_sysman_test.c.

Note: gpu_sysman_test.c needs to use modified simplified versions of most of the collectd functions used by gpu_sysman.c, for test validation purposes.  Linking it with normal collectd code would cause conflicts and collectd code in general does not even build with the GCC warning options I use (with -Werror) for testing the Sysman plugin.

(I have PRs both for v5.x and v6.x collectd branches to fix subset of those warnings.)